ast.go

```diff
@@ -184,6 +280,22 @@ func (ve VariableExpression) String() string {
 	return fmt.Sprintf("{ %s %s }", ve.Variable, ve.Annotation)
 }
 
+func (ve VariableExpression) validate() error {
+	if err := ve.Variable.validate(); err != nil {
```

Comment:
no need to check for ve.Variable being nil?

Reply:
underlying value is string, cant be nil